nixos/modules/services/mail/postal.nix

```diff
@@ -0,0 +1,530 @@
+        via the module options. It should include `RAILS_SECRET_KEY`, which can
+        be generated with `openssl rand -hex 64`.
+        [List of available variables](https://github.com/postalserver/postal/blob/main/doc/config/environment-variables.md)
+      '';
+      example = "/run/secrets/postal.env";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to open the firewall port for the [smtp server](#opt-services.postal.settings.smtp_server.default_port).
+        Note that it won't open the ports for the healthcheck services or the web UI, which should be placed behind a web proxy.
+      '';
+    };
+
+    workers = mkOption {
+      type = types.int;
+      default = 1;
+      description = "Number of workers to handle incoming and outgoing e-mail, to increase if you process lots of e-mails.";
+    };
+
+    settings = lib.mkOption {
+      default = { };
+      description = ''
+        Options for postal configuration. Only [`settings.postal.signing_key_path`](#opt-services.postal.settings.postal.signing_key_path)
+        is mandatory, refer to <https://github.com/postalserver/postal/blob/main/doc/config/yaml.yml> for details on all supported values.
+      '';
+
+      type = lib.types.submodule {
+        freeformType = yaml.type;
+
+        options = {
+          postal = {
+            web_hostname = mkOption {
+              type = types.str;
+              default = cfg.domain;
+              defaultText = literalExpression "config.${opt.domain}";
+              description = "The hostname that the Postal web interface runs on";
+            };
+
+            web_protocol = mkOption {
+              type = types.enum [
+                "http"
+                "https"
+              ];
+              default = "https";
+              description = "The HTTP protocol to use for the Postal web interface";
+            };
+
+            smtp_hostname = mkOption {
+              type = types.str;
+              default = cfg.domain;
+              defaultText = literalExpression "config.${opt.domain}";
+              description = "The hostname that the Postal SMTP server runs on";
+            };
+
+            signing_key_path = mkOption {
+              type = types.path;
+              description = ''
+                Path to the private key used for signing.
+                The web UI won't work without it, can be
+                generated with `openssl genrsa 2048`
+              '';
+            };
+          };
+
+          web_server = {
+            default_port = mkOption {
+              type = types.port;
+              default = 5000;
+              description = "The default port the web server should listen on unless overriden by the PORT environment variable";
+            };
+
+            default_bind_address = mkOption {
+              type = types.str;
+              default = "127.0.0.1";
```

Comment:
We can probably use a unix socket here, and avoid the port collision issue with other modules.

Reply:
It seems postal's [puma config](https://github.com/postalserver/postal/blob/main/config/puma.rb) does not allow to use a socket. I could patch this file in the derivation, and add an option for people who prefer to use port binding instead of the socket. Or maybe we can keep the port for the moment and I'll report/PR upstream to have a proper way to implement the socket option?